File: taskhive-backend/src/main/java/com/digiwork/taskhive/module/analytics/dto/EmployeeDashboardResponse.java
package com.digiwork.taskhive.module.analytics.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EmployeeDashboardResponse {

    private long totalTasks;
    private long todoTasks;
    private long inProgressTasks;
    private long inReviewTasks;
    private long completedTasks;
    private BigDecimal onTimeCompletionRate;
    private BigDecimal avgCompletionHours;
}

File: taskhive_flutter/lib/features/analytics/data/models/employee_dashboard_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'employee_dashboard_response.freezed.dart';
part 'employee_dashboard_response.g.dart';

@freezed
class EmployeeDashboardResponse with _$EmployeeDashboardResponse {
  const factory EmployeeDashboardResponse({
    int? totalTasks,
    int? todoTasks,
    int? inProgressTasks,
    int? inReviewTasks,
    int? completedTasks,
    double? onTimeCompletionRate,
    double? avgCompletionHours,
  }) = _EmployeeDashboardResponse;

  factory EmployeeDashboardResponse.fromJson(Map<String, dynamic> json) =>
      _$EmployeeDashboardResponseFromJson(json);
}
